x-pack/plugins/reporting/server/routes/internal/generate/generate_from_jobparams.ts

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { KibanaResponse } from '@kbn/core-http-router-server-internal';
+import type { Logger } from '@kbn/core/server';
+import type { ReportingCore } from '../../..';
+import { INTERNAL_ROUTES } from '../../../../common/constants';
+import { authorizedUserPreRouting } from '../../common';
+import { RequestHandler } from '../../common/generate';
+
+export function registerGenerationRoutesInternal(reporting: ReportingCore, logger: Logger) {
+  const setupDeps = reporting.getPluginSetupDeps();
+  const { router } = setupDeps;
+
+  // TODO: find a way to abstract this using ExportTypeRegistry: it needs a new
+  // public method to return this array
+  // const registry = reporting.getExportTypesRegistry();
+  // const kibanaAccessControlTags = registry.getAllAccessControlTags();
```

Comment:
I'm not sure if we still care about this, I just copied this comment from `x-pack/plugins/reporting/server/routes/generate/generate_from_jobparams.ts`

Reply:
I am also tempted to remove this comment - I don't think it's really relevant anymore